core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -1571,6 +1569,21 @@ NATable *BindWA::getNATable(CorrName& corrName,
       ((QualifiedName&)(table->getTableName())).setIsVolatile(TRUE);
     }
       
+  // For now, do not allow access through the Trafodion external name created for
+  // the HIVE object unless the inDDL flag is set.  inDDL is set for drop 
+  // table and SHOWDDL statements.  
+  // TDB - may want to merge the Trafodion version with the HIVE version.
+  // TDB - similar operation may be needed for external HBase tables
+  if ((table) && (table->isExternalTable() && (! bindWA->inDDL())))
```

Comment:
inDDL was added for compound create schema commands a while back and does not seem to be used today.  I added it as a stop gate since it was available.  It does make sense to add it as separate fields but I didn't do it because I was not sure if we were going to keep this rule going forward.  We may want to allow all accesses through the external name. I will add this to my list of things to do.